### DISCUSSION TITLE
I'm seeing 429 error for almost 24 hours, any ideas how to fix it?

### DISCUSSION THREAD

**Discussion by zarubin-alexey** [sealbro/dotnet.garmin.connect#69] [Q&A] [answered]
Garmin.Connect.Auth.External.GarminConnectAuthenticationException: 'Garmin Authentication Failed. TooManyRequests: temporary blocked by Garmin' .

I can not find anything about fixing this issue, I'd tried 2 accounts, I can access both via web client, but not using your API library.

@sealbro help me please)

**Comment by sealbro** [comment]
Hi, Did you call `new GarminConnectionClient` per each request or used singleton? 
As a temporary fix/check could you try calling with any proxy?

**Comment by sealbro** [comment, marked as answer]
Since 0.9.0, support for token cache and especially file token cache have been introduced, which will help to make fewer OAuth calls and avoid 429 codes.
